site/src/api/typesGenerated.ts

```diff
@@ -3375,6 +3373,14 @@ export interface WorkspaceAgentDevcontainer {
 	readonly status: WorkspaceAgentDevcontainerStatus;
 	readonly dirty: boolean;
 	readonly container?: WorkspaceAgentContainer;
+	readonly agent?: WorkspaceAgentDevcontainerAgent;
+}
+
+// From codersdk/workspaceagents.go
+export interface WorkspaceAgentDevcontainerAgent {
```

Comment:
This name sounds confusing to me 🤔. From my understanding, we have two agents, WorkspaceAgent, and Devcontainer Agent, is that correct? If yes, why not call DevcontainerAgent to simplify the naming?

Reply:
I do see your point. Technically I suppose "Workspace" in "WorkspaceAgent" is also somehwhat useless. The main idea with this is to follow existing naming conventions but also to symbolize the dependency tree. A devcontainer agent (a workspace agents subagent) cannot exist without a (parent) workspace agent. Hence it is nested in the naming. Another existing type like `WorkspaceAgentContainer` also already exists so I don't think it's worth diverging right now without a plan for how to tackle existing names.